torchgeo/models/weights.py

```diff
@@ -0,0 +1,154 @@
+"""Pretrained weights for models."""
+import os
+from typing import Any, Dict
+
+import torch
+import torch.nn as nn
+from torchvision.models._api import Weights, WeightsEnum
+
+from torchgeo.datasets.utils import download_url
+
+__all__ = ["ResNet50_Weights"]
+
+
+class ResNet50_Weights(WeightsEnum):
+    """ResNet50 weights.
+
+    For `timm
+    <https://github.com/rwightman/pytorch-image-models>`_ implementation.
+    """
+
+    IMAGENET_RGB = Weights(
+        url=(
+            "https://github.com/rwightman/pytorch-image-models/releases/"
+            "download/v0.1-rsb-weights/resnet50_a1_0-14fe96d1.pth"
+        ),
+        transforms=nn.Identity(),
+        meta={
+            "dataset": "imagenet",
+            "ssl_method": None,
+            "repo": (
+                "https://github.com/rwightman/pytorch-image-models/"
+                "blob/main/timm/models/resnet.py"
+            ),
+        },
+    )
+    SENTINEL2_RGB_SECO = Weights(
+        url="https://zenodo.org/record/4728033/files/seco_resnet50_1m.ckpt?download=1",
+        transforms=nn.Identity(),
+        meta={
+            "ssl_method": "seco",
+            "dataset": "seco",
+            "publication": "https://arxiv.org/abs/2103.16607",
+            "repo": "https://github.com/ServiceNow/seasonal-contrast",
+        },
+    )
+
+    AERIAL_MILLIONAID_RGB = Weights(
+        url="https://drive.google.com/file/d/1K3P4_fDfcBRGqpKoSdSa6OXS4xC1xLC9",
+        transforms=nn.Identity(),
+        meta={
+            "dataset": "millionaid",
+            "ssl_method": None,
+            "repo": (
+                "https://github.com/ViTAE-Transformer/"
+                "ViTAE-Transformer-Remote-Sensing"
+            ),
+        },
+    )
+
+    SENTINEL2_ALL_MOCO = Weights(
+        url=(
+            "https://drive.google.com/file/d/1OrtPfG2wkO05bimstQ_T9Dza8z3zp8i-/"
+            "view?usp=sharing/B13_rn50_moco_0099_ckpt.pth"
+        ),
+        transforms=nn.Identity(),
+        meta={
+            "ssl_method": "moco",
+            "publication": "https://arxiv.org/abs/2211.07044",
+            "dataset": "SSL4EO-S12",
+            "repo": "https://github.com/zhu-xlab/SSL4EO-S12",
+        },
+    )
+    SENTINEL2_ALL_DINO = Weights(
+        url="https://drive.google.com/file/d/1iSHHp_cudPjZlshqWXVZj5TK74P32a2q",
+        transforms=nn.Identity(),
+        meta={
+            "ssl_method": "dino",
+            "publication": "https://arxiv.org/abs/2211.07044",
+            "dataset": "SSL4EO-S12",
+            "repo": "https://github.com/zhu-xlab/SSL4EO-S12",
+        },
+    )
+    DEFAULT = SENTINEL2_ALL_MOCO
+
+    def get_state_dict(self) -> Dict[str, Any]:
+        """Retrieve pretrained weights state_dict."""
+        root = os.path.join(os.getcwd(), "weights")
```

Comment:
Probably makes more sense to use the same default as [torchvision](https://pytorch.org/docs/stable/hub.html#torch.hub.load_state_dict_from_url).

Reply:
do you mean naming the default root `checkpoints` instead, that is the default name in torchvision from what I can tell

Reply:
See the docs I linked to above. There is a `model_dir` that lets you override the default, which defaults to `<hub_dir>/checkpoints`, where `<hub_dir>` [depends on OS](https://pytorch.org/docs/stable/hub.html#torch.hub.get_dir)

Reply:
If I use `torch.hub.load_state_dict_from_url()` I get `_pickle.UnpicklingError: invalid load key, '<'.` so it somehow doesn't work with these weights. It does work with the `load_checkpoint_from_url()` function I wrote. So is it okay to use that and then use `torch.hub.get_dir()` as the root where the weights will be placed?

Reply:
Would be worth investigating why the builtin option doesn't work or if we can modify it in some way.

Reply:
ah I think you can only use torch.hub.load_state_dict_from_url() fresh, so not if you have downloaded it before, so deleting the directory did it.

Reply:
the suggestions I see for this is to deleted the directory and redownload it but that doesn't solve it. Maybe it's the way the .pth file was created?

Reply:
maybe this is the issue:
`"https://drive.google.com/file/d/1Tx07L6OilkfcgE2HWiSXHRmRepCPdn6V/view?usp=sharing/B13_vits16_moco_0099_ckpt.pth"` is the link for the download. In the torch source code for `load_checkpoint_from_url()` they use a `urlparse` function which jumps on the `?` after `view` in the url and splits it in such a way that the followed `os.path.basename` is garbage. So will check if I can somehow get different download links?

Reply:
`torchvision.datasets.utils` has a `download_url` function that torchgeo also uses for datasets downloads and it checks for files being from google drive, but I think for `torch.hub.load_checkpoint_from_url()` torch assumes just one source of checkpoint files because they have all their own torchvision models and checkpoints. So I think that is why it does not work with `torch.hub.load_checkpoint_from_url()` but ckpt download does work when using the more generic torchvision ` download_url` function.

Reply:
```
def load_checkpoint_from_url(
    root: str, filename: str, url: str, map_location: torch.device
) -> Dict[str, Any]:
    """Download and load a checkpoint.

    Args:
        root: root directory where checkpoint file should be saved
        filename: filename the downloaded checkpoint file should have
        url: url to checkpoint file to download
        map_location: torch device to load checkpoint

    Returns:
        loaded checkpoint
    """
    if not os.path.exists(os.path.join(root, filename)):
        download_url(url, root=root, filename=filename)
    return torch.load(os.path.join(root, filename), map_location=map_location)
```
This is the function I am using at the moment to download the checkpoints, so it just relies on the `torchvision.datasets` `download_url` function